packages/block-editor/src/components/block-navigation/index.js

```diff
@@ -1,80 +1,30 @@
-/**
- * External dependencies
- */
-import { isArray, noop } from 'lodash';
-
 /**
  * WordPress dependencies
  */
-import { useDispatch, useSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import BlockNavigationTree from './tree';
-import { isClientIdSelected } from './utils';
-import { store as blockEditorStore } from '../../store';
+
+const noop = () => {};
 
 export default function BlockNavigation( {
```

Comment:
It seems we don't really use this component, as we are using the Tree component for the list views.  It also seems this is not exported either.

Im only seeing this used in /block-navigation/dropdown.js ?  Maybe it makes sense to get rid of this component altogether and just update dropdown.js to use Tree appropriately?

It also seems odd for this file to be named `index.js`.  It doesn't look like we export this, im only seeing exports for dropdown, block-slot, editor, and tree specifically.

Reply:
>Im only seeing this used in /block-navigation/dropdown.js ? Maybe it makes sense to get rid of this component altogether and just update dropdown.js to use Tree appropriately?

I agree with this and and since `BlockNavigation` is not exported and only `BlockNavigationDropdown` is, the implementation details are irrelevant as long as it keeps working as before.

Also changes at `Tree` should be okay since it's `__experimental`.

Reply:
I've created a separate PR to address the restructuring, as it would make this extremely confusing to review: https://github.com/WordPress/gutenberg/pull/31892

It branches out from this PR, and only takes care of removing `BlockNavigation` and replacing it with `BlockNavigationTree`.